graphblas/vector.py

```diff
@@ -708,6 +708,49 @@ def apply(self, op, right=None, *, left=None):
             size=self._size,
         )
 
+    def select(self, op, thunk=None):
+        """
+        GrB_Vector_select
+        Compute SelectOp at each element of the calling Vector, keeping
+        elements which return True.
+        """
+        method_name = "select"
+        if thunk is None:
+            thunk = False  # most basic form of 0 when unifying dtypes
+        if type(thunk) is not Scalar:
+            dtype = self.dtype if self.dtype._is_udt else None
```

Comment:
We should only use this pattern if the op is not positional.  A UDT is not the type we want when comparing against indices!